service/worker/esanalyzer/analyzer_test.go

```diff
@@ -0,0 +1,465 @@
+}
+
+func (s *esanalyzerWorkflowTestSuite) TestFindStuckWorkflows() {
+	info := WorkflowTypeInfo{
+		Name:         s.WorkflowType,
+		NumWorkflows: 123113,
+		Duration:     Duration{AvgExecTimeNanoseconds: float64(100 * time.Minute)},
+	}
+
+	s.mockESClient.On("SearchRaw", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(
+		&elasticsearch.RawResponse{
+			TookInMillis: 12,
+			Hits: elasticsearch.SearchHits{
+				TotalHits: 2,
+				Hits: []*persistence.InternalVisibilityWorkflowExecutionInfo{
+					{
+						DomainID:   s.DomainID,
+						WorkflowID: s.WorkflowID,
+						RunID:      s.RunID,
+					},
+					{
+						DomainID:   s.DomainID,
+						WorkflowID: "workflow2",
+						RunID:      "run2",
+					},
+				},
+			},
+		},
+		nil).Times(1)
+	s.mockMetricClient.On(
+		"AddCounter",
+		metrics.ESAnalyzerScope,
+		metrics.ESAnalyzerNumStuckWorkflowsDiscovered,
+		int64(2),
+	).Return().Times(1)
+
+	actFuture, err := s.activityEnv.ExecuteActivity(findStuckWorkflows, info)
+	s.NoError(err)
+	var results []WorkflowInfo
+	err = actFuture.Get(&results)
+	s.NoError(err)
+	s.Equal(2, len(results))
+	s.Equal(WorkflowInfo{DomainID: s.DomainID, WorkflowID: s.WorkflowID, RunID: s.RunID}, results[0])
+	s.Equal(WorkflowInfo{DomainID: s.DomainID, WorkflowID: "workflow2", RunID: "run2"}, results[1])
+}
+
+func (s *esanalyzerWorkflowTestSuite) TestFindStuckWorkflowsNotEnoughWorkflows() {
+	info := WorkflowTypeInfo{
+		Name:         s.WorkflowType,
+		NumWorkflows: int64(s.config.ESAnalyzerMinNumWorkflowsForAvg(s.WorkflowType) - 1),
+		Duration:     Duration{AvgExecTimeNanoseconds: float64(100 * time.Minute)},
+	}
+
+	s.logger.On("Warn", mock.Anything, mock.Anything).Return().Times(1)
+
+	actFuture, err := s.activityEnv.ExecuteActivity(findStuckWorkflows, info)
+	s.NoError(err)
+	var results []WorkflowInfo
+	err = actFuture.Get(&results)
+	s.NoError(err)
+	s.Equal(0, len(results))
+}
+
+func (s *esanalyzerWorkflowTestSuite) TestFindStuckWorkflowsMinNumWorkflowValidationSkipped() {
+	info := WorkflowTypeInfo{
+		Name:         s.WorkflowType,
+		NumWorkflows: int64(s.config.ESAnalyzerMinNumWorkflowsForAvg(s.WorkflowType) - 1),
+		Duration:     Duration{AvgExecTimeNanoseconds: float64(100 * time.Minute)},
+	}
+
+	s.config.ESAnalyzerLimitToTypes = dynamicconfig.GetStringPropertyFn(s.WorkflowType)
+	s.logger.On("Info", mock.Anything, mock.Anything).Return().Times(1)
+	s.mockESClient.On("SearchRaw", mock.Anything, mock.Anything, mock.Anything, mock.Anything).
+		Return(&elasticsearch.RawResponse{}, nil).Times(1)
+
+	actFuture, err := s.activityEnv.ExecuteActivity(findStuckWorkflows, info)
+	s.NoError(err)
+	var results []WorkflowInfo
+	err = actFuture.Get(&results)
+	s.NoError(err)
+	s.Equal(0, len(results))
+}
+
+func (s *esanalyzerWorkflowTestSuite) TestGetWorkflowTypes() {
+	esResult := struct {
+		Buckets []WorkflowTypeInfo
```

Comment:
fwiw: this will serialize as `{"Buckets": ...}` but the workflow code is expecting `{"buckets": ...}`.
Which apparently doesn't matter - TIL / I'm very surprised it's case-insensitive, since that's not language-agnostic and it writes with the same value as the field name 🤨